pyquil/operator_estimation.py

```diff
@@ -814,11 +816,102 @@ def serializable(self):
+        estimated and also match the symmetrization type used to estimate the observables of the
+        input ExperimentResults.
+    :param noisy_program: an optional program from which to inherit a noise model; only relevant
+        for running on a QVM
+    :param active_reset: whether or not to begin the program by actively resetting. If true,
+        execution of each of the returned programs in a loop on the QPU will generally be faster.
+    :param show_progress_bar: displays a progress bar if true.
+    :return: a copy of the input results with updated estimates and calibration results.
+    """
+    # get unique observables that will need to be calibrated
+    observables = {copy(res.setting.out_operator) for res in expt_results}
+
+    calibrations = {}
+    for obs in tqdm(observables, disable=not show_progress_bar):
+        prog = get_calibration_program(obs, noisy_program, active_reset)
+        meas_qs = obs.get_qubits()
+        results = qc.run_symmetrized_readout(prog, n_shots, symm_type, meas_qs)
+
+        # Obtain statistics from result of experiment
+        # TODO: we have to fabricate an ExperimentSetting to pass to _stats_from_measurements
+        #  even though it only needs the observable.
+        setting = ExperimentSetting(zeros_state(meas_qs), obs)
+        obs_mean, obs_var = _stats_from_measurements(results,
+                                                     {q: idx for idx, q in enumerate(meas_qs)},
+                                                     setting, len(results))
+        calibrations[obs.operations_as_set()] = (obs_mean, obs_var, len(results))
+
+    for expt_result in expt_results:
+        # TODO: allow weight > symm_type
+        if -1 < symm_type < len(expt_result.setting.out_operator.get_qubits()):
+            warnings.warn(f'Calibration of observable {expt_result.setting.out_operator} '
+                          f'currently not supported since it acts on more qubits than the '
+                          f'symm_type {symm_type}.')
+
+        # get the calibration data for this observable
+        cal_data = calibrations[expt_result.setting.out_operator.operations_as_set()]
+        obs_mean, obs_var, counts = cal_data
+
+        # Use the calibration to correct the mean and var
+        result_mean = expt_result.expectation
+        result_var = expt_result.std_err ** 2
+        corrected_mean = result_mean / obs_mean
+        corrected_var = ratio_variance(result_mean, result_var, obs_mean, obs_var)
+
+        yield ExperimentResult(
+            setting=expt_result.setting,
+            expectation=corrected_mean,
+            std_err=np.sqrt(corrected_var),
+            total_counts=expt_result.total_counts,
+            raw_expectation=result_mean,
+            raw_std_err=expt_result.std_err,
+            calibration_expectation=obs_mean,
+            calibration_std_err=np.sqrt(obs_var),
+            calibration_counts=counts
```

Comment:
It's mostly just copied from the old code below, but since this is its own method it makes sense to drop the `calibr_` prefix from the variable names. I didn't think there would be confusion because all of the old stuff is accessed from the `expt_result` being calibrated. 

1) update the `expectation` and `std_err` with the corrected versions. 
2) repopulate the `setting`, `total_counts`, and `raw_...` with the original data from the pre-calibrated `ExperimentResult`
3) Store the calibration data for this observable in the last three fields.

```python
    yield ExperimentResult(
        setting=setting,
        expectation=corrected_mean.item(),
        std_err=np.sqrt(corrected_var).item(),
        total_counts=n_shots,
        raw_expectation=obs_mean.item(),
        raw_std_err=np.sqrt(obs_var).item(),
        calibration_expectation=obs_calibr_mean.item(),
        calibration_std_err=np.sqrt(obs_calibr_var).item(),
        calibration_counts=calibr_shots,
    )
```